File: python/personal_projects/Others/calculator.py
# Simple calculator with python

# operation functions
def sum(x, y):
    return x + y

def substract(x, y):
    return x - y

def multiply(x, y):
    return x * y

def divide(x, y):
    return x / y

print("Select operation:")
print("1.Sum")
print("2.Substract(çıkarma)")
print("3.Multiply(çarpma)")
print("4.Divide(bölme)")

while True:
    # Take input from the user
    choice = input("Enter choice(1,2,3,4): ")

    # check if choice is one of the four options
    if choice in ("1", "2", "3", '4'):    
        # example: float = 2.0, int = 2       
        num1 = float(input("Enter first number: "))
        num2 = float(input("Enter second number: "))
        

        if choice == "1":
            print(num1, "+", num2, "=", sum(num1, num2))
        elif choice == "2":
            print(num1, "-", num2, "=", sum(num1, num2))
        elif choice == "3":
            print(num1, "*", num2, "=", sum(num1, num2))
        elif choice == "4":
            print(num1, "/", num2, "=", sum(num1, num2))
        
        # check if user wants another calculation
        # break the wile loop if answer is no
        next_calculation = input("Let's do next calculation? (y/n): ")
        if next_calculation == "n":
            break
    else:
        print("Invalid input")
        break

File: C/CS50x_Computer-Science/Lecture-1/calculator.c
#include <stdio.h>

int main(void)
{
    // calculator app
    printf("Welcome to the calculator app!");
    // select case
    printf("Select an operation: 1. Add 2. Subtract 3. Multiply 4. Divide: ");
    int operation;
    scanf("%d", &operation);
    // get numbers
    printf("Enter number 1: ");
    float num1, num2;
    scanf("%f", &num1);
    printf("Enter number 2: ");
    scanf("%f", &num2);
    // perform operation
    float result;

    switch (operation)
    {
        case 1:
            result = num1 + num2;
            printf("Result: %f", result);
            break;
        case 2:
            result = num1 - num2;
            printf("Result: %f", result);
            break;
        case 3:
            result = num1 * num2;
            printf("Result: %f", result);
            break;
        case 4:
            result = num1 / num2;
            printf("Result: %f", result);
            break;
        default:
            printf("Invalid operation!");
            break;
    }

    printf("\n");
    return 0;
    
}